crates/bevy_math/src/curve/mod.rs

```diff
@@ -968,19 +971,135 @@ where
     }
 }
 
+/// Convert a `start` and `end` point of some [`VectorSpace`] type into a [`Curve`] over the
+/// [unit interval], sampled by linearly interpolating between the two boundary values.
+///
+/// [unit interval]: `Interval::UNIT`
+pub fn line_curve<T: VectorSpace>(start: T, end: T) -> impl Curve<T> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: move |t: f32| start * (1.0 - t) + end * t,
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(x) = x²`
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn quadratic_ease_in() -> impl Curve<f32> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: |t: f32| t.squared(),
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(x) = 1 - (1 - x)²`
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn quadratic_ease_out() -> impl Curve<f32> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: |t: f32| 1.0 - (1.0 - t).squared(),
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(x) = x² * (3 - 2x)`
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn cubic_curve() -> impl Curve<f32> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: |t: f32| t.squared() * (3.0 - 2.0 * t),
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(n,x) = round(x * n) / n`
+///
+/// parametrized by `n`, the number of steps
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn step_curve(num_steps: usize) -> impl Curve<f32> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: move |t: f32| (t * num_steps as f32).round() / num_steps as f32,
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(omega,x) = (1 - (1 - x)²)(2sin(omega * x) / omega + cos(omega * x))`
+///
+/// parametrized by `omega`
+///
+/// [unit domain]: `Interval::UNIT`
```

Comment:
As far as I can tell, for most values of omega, this method does not actually map the interval to itself, furthermore, I don't see how this function specifically is useful, what is it trying to achieve, maybe we can tweak it.